lib/elixir/lib/io/ansi/docs.ex

```diff
@@ -366,6 +366,10 @@ defmodule IO.ANSI.Docs do
     process_fenced_code_block(rest, text, indent, options, _delimiter = "~~~")
   end
 
+  defp process(["<!--" <> _line | rest], text, indent, options) do
+    process(rest, text, indent, options)
```

Comment:
this implementation is very simplistic. It assumes the line starting with `<!--` ends with that too, which may not always be the case. I'm happy to make it more robust, including properly handling inline comments if that seems important.